lldb/test/API/python_api/find_in_memory/address_ranges_helper.py

```diff
@@ -0,0 +1,31 @@
+import lldb
+
+SINGLE_INSTANCE_PATTERN = "there_is_only_one_of_me"
+DOUBLE_INSTANCE_PATTERN = "there_is_exactly_two_of_me"
+
+
+def GetAddressRanges(test_base):
```

Comment:
Here you can just get the current thread's stack pointer for "there_is_only_one_of_me" and then get its region, and using the SBValue for "str1" and "str2" (see changes to the test program below), you can extract the pointer to the heap copy, and then use that address region.

Reply:
Why not just store them all on stack and only use one region to search through?

Reply:
You will want to check that multiple memory ranges works, so you need at least two right?